diff --git a/sims/vcs/Makefile b/sims/vcs/Makefile
index 8517fc1d..f942a17b 100644
--- a/sims/vcs/Makefile
+++ b/sims/vcs/Makefile
@@ -69,7 +69,7 @@ model_dir_debug = $(build_dir)/$(long_name).debug
 #########################################################################################
 $(sim): $(sim_common_files) $(dramsim_lib) $(EXTRA_SIM_REQS)
 	rm -rf $(model_dir)
-	$(VCS) $(VCS_OPTS) $(EXTRA_SIM_SOURCES) -o $@ -Mdir=$(model_dir)
+	$(VCS) $(VCS_OPTS) $(EXTRA_SIM_SOURCES) -o $@ -Mdir=$(model_dir) -lca -cm line+cond+fsm+tgl+path+branch+assert -cm_dir Coverage.vdb
 
 $(sim_debug): $(sim_common_files) $(dramsim_lib) $(EXTRA_SIM_REQS)
 	rm -rf $(model_dir_debug)
